web/modules/custom/form_api/src/Form/UserForm.php

```diff
@@ -0,0 +1,236 @@
+      '#states' => [
+        'visible' => [
+          ':input[id="gender"]' => ['value' => 'other'],
+        ],
+      ],
+    ];
+    $form['actions'] = [
+      '#type' => 'submit',
+      '#value' => $this->t('Submit'),
+      '#ajax' => [
+        'callback' => '::submitUsingAjax',
+        'progress' => [
+          'type' => 'throbber',
+          'message' => NULL,
+        ],
+      ],
+      '#suffix' => "<div id='submitted'></div>",
+    ];
+
+    return $form;
+  }
+
+  /**
+   * Validates the form using AJAX and submits it.
+   *
+   * @param array $form
+   *   An associative array containing the structure of the form.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The current state of the form.
+   *
+   * @return \Drupal\Core\Ajax\AjaxResponse
+   *   The AJAX response.
+   */
+  public function submitUsingAjax(array &$form, FormStateInterface $form_state) {
+    $error_array = $this->validateFormUsingAjax($form_state);
+    $triggering_element = $form_state->getTriggeringElement();
+    $response = new AjaxResponse();
+    if (count($error_array) == 0  && $triggering_element['#type'] === 'submit') {
+      $config = $this->configFactory()->getEditable('form_api.settings');
+      $config->set('full_name', $form_state->getValue('full_name'));
+      $config->set('phone_number', $form_state->getValue('phone_number'));
+      $config->set('email', $form_state->getValue('email'));
+      $config->set('gender', $form_state->getValue('gender'));
+      $config->save();
+      $message = $this->t('Thanks! For Submitting The Form.');
+      $response->addCommand(new HtmlCommand('#submitted', $message));
+    }
+    else {
+      $css_string = '<style>.red{color:red;}</style>';
+      if (array_key_exists('full-name-result', $error_array)) {
+        $response->addCommand($error_array['full-name-result']);
+      }
+      if (array_key_exists('phone-number-result', $error_array)) {
+        $response->addCommand($error_array['phone-number-result']);
+      }
+      if (array_key_exists('email-result', $error_array)) {
+        $response->addCommand($error_array['email-result']);
+      }
+      $response->addCommand(new AddCssCommand($css_string));
+    }
+    return $response;
+  }
+
+  /**
+   * Validates the form and returns AJAX response.
+   *
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The current state of the form.
+   *
+   * @return \Drupal\Core\Ajax\HtmlCommand[]
+   *   The array containing all error messages.
+   */
+  public function validateFormUsingAjax(FormStateInterface $form_state) {
+    $phone_number = $form_state->getValue('phone_number');
+    $email_value = $form_state->getValue('email');
+    $full_name = $form_state->getValue('full_name');
+    $email_validator = $this->emailValidator->isValid($email_value);
+    $email_providers = [
+      'gmail.com',
+      'yahoo.com',
+      'outlook.com',
+      'hotmail.com',
+      'mail.com',
+      'zoho.com',
```

Comment:
This should come from config.
What if tomorrow I want to add/remove from this list?
If this is part of code, then I have to raise PR, merge and wait for code to move to live.
If this is part of config, then I can easily edit the config in live using drush.

Reply:
Okay, implemented using config.